packages/in_app_purchase/in_app_purchase_storekit/darwin/Classes/InAppPurchasePlugin.m

```diff
@@ -265,35 +244,27 @@ - (void)finishTransaction:(FlutterMethodCall *)call result:(FlutterResult)result
       @try {
         [self.paymentQueueHandler finishTransaction:transaction];
       } @catch (NSException *e) {
-        result([FlutterError errorWithCode:@"storekit_finish_transaction_exception"
-                                   message:e.name
-                                   details:e.description]);
+        *error = [FlutterError errorWithCode:@"storekit_finish_transaction_exception"
+                                     message:e.name
+                                     details:e.description];
         return;
       }
     }
   }
-
-  result(nil);
 }
 
-- (void)restoreTransactions:(FlutterMethodCall *)call result:(FlutterResult)result {
-  if (call.arguments && ![call.arguments isKindOfClass:[NSString class]]) {
-    result([FlutterError
-        errorWithCode:@"storekit_invalid_argument"
-              message:@"Argument is not nil and the type of finishTransaction is not a string."
-              details:call.arguments]);
-    return;
-  }
-  [self.paymentQueueHandler restoreTransactions:call.arguments];
-  result(nil);
+- (void)restoreTransactionsApplicationUserName:(nullable NSString *)applicationUserName
+                                         error:(FlutterError *_Nullable __autoreleasing *_Nonnull)
+                                                   error {
+  [self.paymentQueueHandler restoreTransactions:applicationUserName];
 }
 
+- (void)presentCodeRedemptionSheetWithError:
+    (FlutterError *_Nullable __autoreleasing *_Nonnull)error {
 #if TARGET_OS_IOS
-- (void)presentCodeRedemptionSheet:(FlutterMethodCall *)call result:(FlutterResult)result {
   [self.paymentQueueHandler presentCodeRedemptionSheet];
-  result(nil);
```

Comment:
I just want to call out how these calls have disappears now.  In the past, failing to call these created bugs, you've eliminated that risk now 👍